ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -75,14 +77,14 @@ method close*(subscriptions: JsonRpcSubscriptions) {.async, base.} =
     await subscriptions.unsubscribe(id)
 
 proc getCallback(subscriptions: JsonRpcSubscriptions,
-                 id: JsonNode): ?SubscriptionCallback =
+                 id: JsonNode): ?SubscriptionCallback  {. raises:[].} =
   try:
     if not id.isNil and id in subscriptions.callbacks:
-      subscriptions.callbacks[id].some
-    else:
-      SubscriptionCallback.none
+      try:
+        return subscriptions.callbacks[id].some
+      except: discard
   except KeyError:
-    SubscriptionCallback.none
+    return SubscriptionCallback.none
```

Comment:
`KeyError` will not leak from this, so no need to catch it. The entire block can be simplified to:
```nim
if not id.isNil and id in subscriptions.callbacks:
  try:
    return subscriptions.callbacks[id].some
  except: discard
```
OR
```nim
try:
  if not id.isNil and id in subscriptions.callbacks:
    return subscriptions.callbacks[id].some
except: discard
```